### DISCUSSION TITLE
UX Suggestion

### DISCUSSION THREAD

**Discussion by davidbdyer** [massCodeIO/massCode#275] [Ideas]
I saw your note on the download page of your site. This might be part of why you are not gaining traction, idk. I tried MassCode on Mac OS to see what was out there as an alternative to Snippet Labs. So this only applies to Mac OS. Once I saw it put a visible folder in my home directory, I immediately deleted it. I didn't look at it again until my Setapp subscription was about to run out two days ago. I immediately deleted the folder it created. Putting visible folders in a user's home directory on Mac OS goes against established UX. If my idea offends just delete it.

**Comment by antonreshetov** [comment]
First of all, I have not heard before that placing files in the user's home directory is a violation of anything. I'm sorry, where should the user's files be placed? You installed applications under your user. Can you give me a link where you can read this?

Secondly, I do not think that the low popularity of the project has anything to do with what you wrote. If it was, you obviously would not have been among the first who wrote about it.

**Comment by davidbdyer** [comment]
I would start with Apple's documentation for best practices. I don't install applications under my user. The default install location is "/Applications". Mac OS has a Library in the user directory, and most apps store files in Application Support. A few will store files in the user directory, mostly command line tools, but they are hidden by prefixing the file or folder with a "." If every app visibly stored files or folders in the root of the home folder, it would become cluttered and difficult to use. I assume most people don't bother to say anything because of the defensive attitude they receive as a thank you. Which is why I ended my suggestion with "If my idea offends just delete it."

**Reply by antonreshetov** [reply]
I asked for the purpose of self development, I am a macOS user but I have not developed native applications for the OS. Thank you I will know to store application files `/Users/<name>/Library/Application Support/<appName>`.

**Reply by antonreshetov** [reply]
I will change the default folder for storing files in a future version. But you can still test the application, I do not think the current behavior of the application, it is a big deal that prevents it.

**Comment by davidbdyer** [comment]
I was confused at first by the inbox folder. It made me think the app had email integration.

I like the theme selection and font choice, and overall, MassCode looks better to me than Snippet Labs. I'm happy to see Raycast integration.
